src/swarm-tui/Swarm/TUI/View/Robot.hs

```diff
@@ -125,30 +270,38 @@ robotsListWidget s = hCenter table
       | otherwise = show (age `div` 3600 * 24) <> "day"
      where
       TimeSpec createdAtSec _ = robot ^. robotCreatedAt
-      TimeSpec nowSec _ = s ^. uiState . uiGameplay . uiTiming . lastFrameTime
+      TimeSpec nowSec _ = c ^. timing . lastFrameTime
       age = nowSec - createdAtSec
 
     rInvCount = sum $ map fst . E.elems $ robot ^. robotEntity . entityInventory
     rLog
       | robot ^. robotLogUpdated = "x"
       | otherwise = " "
 
-    locWidget = hBox [worldCell, str $ " " <> locStr]
+    locWidget =
+      WidthWidget (2 + length locStr) w
      where
+      w = hBox [worldCell, str $ " " <> locStr]
       rCoords = fmap locToCoords rLoc
       rLoc = robot ^. robotLocation
       worldCell =
         drawLoc
-          (s ^. uiState . uiGameplay)
+          (c ^. gameplay)
           g
           rCoords
       locStr = renderCoordsString rLoc
 
     statusWidget = case robot ^. machine of
-      Waiting {} -> txt "waiting"
+      Waiting {} ->
+        let tx = "waiting"
+         in WidthWidget (length tx) (str tx)
       _
-        | isActive robot -> withAttr notifAttr $ txt "busy"
-        | otherwise -> withAttr greenAttr $ txt "idle"
+        | isActive robot ->
+            let tx = "busy"
+             in WidthWidget (length tx) (withAttr notifAttr $ str tx)
```

Comment:
`WidthWidget` should be a `Functor` in the second field.  Then should re-use the local `strWidget` function here.